### ISSUE TITLE
Extend connect API C++ to handle Viewer's later start or reconnect

### ISSUE THREAD

**Issue by rgolovanov** [rerun-io/rerun#7090]
Currently, the workflow of the logger-viewer implies that the viewer is running before logging starts, otherwise, it will lead to data accumulation (see #6933 ) and never connect. In practice, it is important to have the ability to know whether `connect` to the viewer was successful or not and monitor connectivity status somehow.

**Describe the solution you'd like**
* Extend C++ API to indicate whether connect is successful or not - this would allow users to implement logic which attempts to connect.
* Extend C++ API to indicate that the Viewer has been disconnected - this would allow users to handle such cases and initiate reconnect.

**Comment by Wumpf** [comment]
I don't think this applies to the C++ sdk in particular and is rather an issue in all APIs

**Comment by alokpr** [comment]
This feature would be very useful. Currently the SDK does not even provide connection status that clients can use to handle this on their end.

**Comment by WorstCodeWay** [comment]
Hi, is this issue still active? Thanks.

**Comment by WorstCodeWay** [comment]
There is a server mode in rerun sdk. In such mode, we can later open a rerun viewer like, `rerun --connect rerun+http://127.0.0.1:9876/proxy` to connect to a exist rerun server. Note please, `127.0.0.1` should be your correct ip address rerun server running on.

This feature meets my need. Codes as following:

```
      // Parse address to extract host and port
      std::string host = "0.0.0.0";
      uint16_t port = 9876;

      auto result = rec_->serve_grpc(host, port);
```

**Edit:**
Be careful, it looks like the memory usage of the Rerun server keeps increasing until it eventually OOMs.
